lightning/src/offers/signer.rs

```diff
@@ -410,6 +410,6 @@ pub(crate) fn hmac_for_payment_id(
 
```

Comment:
Re: `hmac_for_offer` call above -- it looks like `ExpandedKey::hmac_for_offer` docs need an update since it is only supposed to be used to construct offer metadata atm. It almost looks like we should have used a new key for `InvoiceError` verification, though I might be missing a prior discussion.

Reply:
Right, we are using it for both offer metadata and payer metadata. I think we were fine with this because we add different inputs for each use case. @TheBlueMatt Is that correct?